worker-queue-rabbit/src/main/java/com/github/workerframework/queues/rabbit/WorkerQueueConsumerImpl.java

```diff
@@ -126,14 +131,28 @@ public void processDelivery(Delivery delivery)
             } catch (final CodecException e) {
                 throw new InvalidDeliveryException("Cannot deserialize delivery messageData to TaskMessage", inboundMessageId, e);
             }
-            handleTaskDataInjection(taskMessage, inboundMessageId, taskMessageStorageRefOpt);
+
+            try {
+                handleTaskDataInjection(taskMessage, inboundMessageId, taskMessageStorageRefOpt);
```

Comment:
The ReferenceNotFoundException and InvalidDeliveryException are both terminal problems that should fail the job. 

Remove the ReferenceNotFoundException from the throws list of handleTaskDataInjection and update the catch InvalidDeliveryException block to send a tracking report.